### PR TITLE
MAT-7951 clearing test case groups for qdm measures when scoring or patient basis is changed

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-7951](https://jira.cms.gov/browse/MAT-7951)
(Optional) Related Tickets:

### Summary
clearing test cases for qdm measures when scoring or patient basis is changed

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java [hunk 1]
```diff
@@ -252,6 +252,28 @@ public Measure updateMeasure(
       // prevent users from manually clearing errors!
       outputMeasure.setErrors(existingMeasure.getErrors());
     }
+
+    // clear testcase data for qdm when scoring or patient basis is changed
+    // for QDM scoring and patient basis are present outside the group
+    // therefor we need to clear testcases while updating measure
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/MeasureService.java]:
```suggestion
    // clear testcase groups for qdm when scoring or patient basis is changed.
    // for QDM, scoring and patient basis are present outside the group
    // therefor we need to clear testcase groups while updating measure
```

**Review by adongare — DISMISSED** [review]

**Review by adongare — APPROVED** [review]
